scripts/ccpp_track_variables.py

```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/env python
+
+# Standard modules
+import argparse
+import logging
+import collections
+import glob
+
+# CCPP framework imports
+from metadata_table import find_scheme_names, parse_metadata_file
+from ccpp_prebuild import import_config, gather_variable_definitions
+from mkstatic import Suite
+from parse_checkers import registered_fortran_ddt_names
+
+###############################################################################
+# Set up the command line argument parser and other global variables          #
+###############################################################################
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-s', '--sdf',           action='store', \
+                    help='suite definition file to parse', required=True)
+parser.add_argument('-m', '--metadata_path', action='store', \
+                    help='path to CCPP scheme metadata files', required=True)
+parser.add_argument('-c', '--config',        action='store', \
+                    help='path to CCPP prebuild configuration file', required=True)
+parser.add_argument('-v', '--variable',      action='store', \
+                    help='variable to track through CCPP suite', required=True)
+#parser.add_argument('--draw',  action='store_true', \
+#                    help='draw graph of calling tree for given variable', default=False)
+parser.add_argument('--debug', action='store_true', help='enable debugging output', default=False)
+args = parser.parse_args()
+
+###############################################################################
+# Functions and subroutines                                                   #
+###############################################################################
+
+def parse_arguments(args):
+    """Parse command line arguments."""
+    success = True
+    sdf = args.sdf
+    var = args.variable
+    configfile = args.config
+    metapath = args.metadata_path
+    debug = args.debug
+    return(success,sdf,var,configfile,metapath,debug)
```

Comment:
The value of `success` never changes here. 

Main and parse args are going to be automatically coupled. Why not just use your `args` Namespace directly in the `main` function?

I think that this is a level of duplication (and indirection) that is going to lead to problems for you down the road. Any time you add or modify a command line argument for this script, you'll need to change the interface for this function that doesn't actually buy you much in the way of decoupling from a Namespace data structure.